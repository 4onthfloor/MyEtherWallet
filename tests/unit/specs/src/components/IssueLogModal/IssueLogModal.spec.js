import Vue from 'vue';
import IssueLogModal from '@/components/IssueLogModal/IssueLogModal.vue';
import { shallowMount } from '@vue/test-utils';
import { Tooling } from '@@/helpers';
import StandardButton from '@/components/Buttons/StandardButton';
import sinon from 'sinon';
import MenuTitle from '@/components/MenuTitle';
import PopOver from '@/components/PopOver';
import { actions } from '@@/helpers/mockStore';
import VueX from 'vuex';

const hideModal = sinon.stub();

const BModalStub = {
  name: 'b-modal',
  template: '<div><slot></slot></div>',
  props: ['to'],
  methods: {
    hide: hideModal
  }
};

const BBtnStub = {
  name: 'b-btn',
  template: '<div class="b-btn">{{id}}</div>',
  props: ['id']
};

const BColStub = {
  name: 'b-col',
  template: '<div class="b-col"><slot></slot></div>',
  props: ['id']
};

const eventHub = {
  $on: sinon.stub()
};

describe('IssuesLogModal.vue', () => {
  let localVue, i18n, wrapper, store;

  beforeAll(() => {
    const baseSetup = Tooling.createLocalVueInstance();
    localVue = baseSetup.localVue;
    i18n = baseSetup.i18n;
    store = store = new VueX.Store({
      modules: {
        main: {
          namespaced: true,
          actions
        }
      }
    });
    Vue.config.warnHandler = () => {};
  });

  beforeEach(() => {
    wrapper = shallowMount(IssueLogModal, {
      localVue,
      i18n,
      store,
      attachToDocument: true,
      stubs: {
        'menu-title': MenuTitle,
        popover: PopOver,
        'b-col': BColStub,
        'b-btn': BBtnStub,
        'b-modal': BModalStub,
        'standard-button': StandardButton
      },
      mocks: {
        $eventHub: eventHub
      }
    });
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correct showSkipper data', async () => {
    wrapper.setData({ showSkipper: true });
    await wrapper.vm.$nextTick();
    expect(wrapper.find('.button-block').exists()).toBe(true);
    expect(wrapper.find('.custom-marker').classes()).not.toContain('enable');
    wrapper.setData({ neverShow: true });
    await wrapper.vm.$nextTick();
    expect(wrapper.find('.custom-marker').classes()).toContain('enable');
  });

  describe('IssueLogModal.vue Methods', () => {
    it('should switch neverShow data when button clicked', async () => {
      wrapper.setData({ showSkipper: true });
      await wrapper.vm.$nextTick();
      wrapper.find('.checkbox-container label').trigger('click');
      expect(wrapper.vm.$data.neverShow).toBe(true);
      wrapper.find('.checkbox-container label').trigger('click');
      expect(wrapper.vm.$data.neverShow).toBe(false);
    });
  });

  it('should dismount properly', () => {
    expect(wrapper.destroy()).toBe(undefined);
  });
});
