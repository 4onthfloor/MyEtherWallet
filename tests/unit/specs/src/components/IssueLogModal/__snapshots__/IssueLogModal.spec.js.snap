// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IssuesLogModal.vue should render correctly 1`] = `
<div class="wrap">
  <div class="modal-container">
    <div hide-footer="" hide-header="" centered="" static="" lazy="" class="bootstrap-modal nopadding">
      <div class="new-issue-log">
        <div class="large-header">Oops</div>
        <div class="sub-header">
          Something went wrong. You can help us fix it by submitting a report about this error.
        </div>
        <div class="buttons mt-5">
          <div class="button-block-text">
            Do you want to inform MEW about this error?
          </div>
          <div class="mt-3 d-flex">
            <div class="b-btn mr-1"></div>
            <div class="b-btn" variant="outline-secondary"></div>
          </div>
          <!---->
          <div class="mt-5 show-error-detail" aria-controls="collapse-error-detail" aria-expanded="false" role="button">
            Show error details
          </div>
        </div>
        <b-collapse-stub id="collapse-error-detail" tag="div" class="mt-2">
          <b-card-stub bodytag="div" title="" titletag="h4" subtitle="" subtitletag="h6" subtitletextvariant="muted" headertag="div" footertag="div" tag="div"><textarea class="error-detail"></textarea></b-card-stub>
        </b-collapse-stub>
      </div>
    </div>
  </div>
</div>
`;
